src/api/apiService.js

```diff
@@ -1,15 +1,54 @@
-import { DEPARTMENT_BY_ID_URL, DEPARTMENTS_URL, LOGIN } from '../constants/url';
+import {
+  DEPARTMENT_BY_ID_URL,
+  DEPARTMENTS_URL,
+  LOGIN_URL,
+  LOGOUT_URL,
+  EMPLOYEES_URL,
+} from '../constants/url';
 import { api } from './interceptors';
 
 export default class apiService {
-  getDepartments() {
-    return api.get(DEPARTMENTS_URL);
+  getDepartments(limit = 10, page = 1) {
+    return api.get(`${DEPARTMENTS_URL}?limit=${limit}&page=${page}`);
   }
-  getDepartmentById(id) {
-    return api.get(`${DEPARTMENT_BY_ID_URL}${id}`);
+  getDepartment(id) {
+    return api.get(`${DEPARTMENTS_URL}/${id}`);
+  }
+  getSearchingDepartment(search) {
+    return api.get(`${DEPARTMENTS_URL}?q=${search}`);
+  }
+  getSortedDepartments(order) {
+    return api.get(`${DEPARTMENTS_URL}?sort=${order}`)
+  }
+  getEmployeesByDepartmentId(id) {
+    return api.get(`${DEPARTMENT_BY_ID_URL}${id}${EMPLOYEES_URL}`);
+  }
+  getEmployee(id) {
+    return api.get(`${EMPLOYEES_URL}/${id}`);
+  }
+  addDepartment(departmentData) {
+    return api.post(DEPARTMENTS_URL, departmentData);
+  }
+  addEmployee(employeeData) {
+    return api.post(EMPLOYEES_URL, employeeData);
```

Comment:
also using constants in a such way might complexify the reverse engineering. (it is much simpler to search for the /specific/endpoint/path than by searching through the constants chain.

But no need to change though